### PR TITLE
MeanFlows implementation.

### PR DESCRIPTION
**Pull request by gumran**
A small implementation of _MeanFlows_ based on the [original paper](https://arxiv.org/pdf/2505.13447).

The notebook is an adjusted version of the `2d_flow_matching.ipynb` notebook. We double the hidden layer size, the number of hidden layers in the MLP and the number of training iterations. The learning rate is reduced to 3e-4. 

The training algorithm is based on Algorithm 1 from the paper, using `torch.func.jvp` to predict the average velocity. We use adaptive loss weighting like the authors do with _p_ = 1 (see their Table 1d), and we set 75% of `r` and `t` pairs to be equal, which corresponds to vanilla flow matching (see Table 1a). We also add gradient clipping.

We use the same convention as in the paper: `x_0` is an image and `x_1` is noise. To sample using _MeanFlows_, we subtract from noise the average velocity from 0 to 1 as predicted by the neural net.

The one-step sampling performance shown in the notebook seems to be about as good as it gets with the given MLP architecture.  We can further double the abovementioned hyperparameters, but so far it has been problematic with my compute restrictions.

### CODE REVIEW

File: examples/2d_mean_flow.ipynb [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+      "source": [
+        "import time\n",
+        "import torch\n",
+        "\n",
+        "from torch import nn, Tensor\n",
+        "\n",
+        "# flow_matching\n",
+        "from flow_matching.path.scheduler import CondOTScheduler\n",
+        "from flow_matching.path import AffineProbPath\n",
+        "from flow_matching.solver import Solver, ODESolver\n",
```
[the remark this answers is not in the document]
Comment (gumran) [reply examples/2d_mean_flow.ipynb:54]: Only the `solver` imports are unused.

**Comment by itaigat** [comment]
Thanks @gumran! Added Mingyang (@lambertae) to review.

File: examples/2d_mean_flow.ipynb [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+        "\n",
+        "    # sample data (user's responsibility): in this case, (X_0,X_1) ~ pi(X_0,X_1) = q(X_1)N(X_0|0,I)\n",
+        "    x_0 = inf_train_gen(batch_size=batch_size, device=device) # sample data\n",
+        "    x_1 = torch.randn_like(x_0).to(device)\n",
+        "\n",
+        "    # sample two time points from a logit-normal distribution\n",
+        "    r = torch.sigmoid(torch.randn(x_1.shape[0], device=device) - 0.4)\n",
+        "    t = torch.sigmoid(torch.randn(x_1.shape[0], device=device) - 0.4) # mean -0.4, var 1\n",
+        "\n",
+        "    # set r = t for 75% of the batch\n",
```
Comment (amorehead) [inline examples/2d_mean_flow.ipynb]: I believe the Mean Flows paper says "The ratio of sampling r != t is 75%". In which case, shouldn't `r = t` 25% of the time?
